### DISCUSSION TITLE
GROTT data to MQTT docker synology nas

### DISCUSSION THREAD

**Discussion by Winfryd** [johanmeijer/grott#584] [Q&A]
I have no luck sending data from grott to mqtt. In the grott terminal I see all the right data coming by but is not send to MQTT. In the log I have this stuborn entry on the second line.

- Grott MQTT topic used : energy/growatt                                
         - MQTT send failed: [Errno 99] Cannot assign requested address          
         - Grott Send data to PVOutput disabled                                  
         - Grott Send data to Influx disabled                                    
         - Grott extension processing disabled                                   
                                                                                 
         - Growatt packet received:                                              
                  <socket.socket fd=5, family=2, type=1, proto=0, laddr=('172.17.0.2', 5279), raddr=('172.17.0.1', 44060)>                                        
         - Data less then minimum record length, data not processed  >        

 I searched the internet for clues and tried a lot of things to no avail. I have the idea that grott.ini is not read but no idea how to check this. The run command is ' python -u grott.py -v  ' but can not find to change this in docker. These are the enviromentals:
                               
        
![image](https://github.com/user-attachments/assets/4ea53977-7ebe-48a0-8c7f-d9e71b98b86e)

Can anyone shed some light on this? I would be highly appreciated. Thank you.

**Comment by Winfryd** [comment]
![image](https://github.com/user-attachments/assets/47a9894a-eaee-4d9f-a0d6-f636bb6f78ca)

Domoticz does connect to mqtt

**Comment by johanmeijer** [comment]
It looks to me that within grott the MQTT adres is not ok.  Without any configuration Grott expect the IP address of MQTT and Grott are the same. Using containers (docker, NAS) each container (and application) has its own IP address. So please specify in the grott.ini (I see you can use this one) in the [MQTT} part IP=172.17.0.3. I am not sure if this very sustainable (are the IP addresses stable or differ they per new run command?). 

Other possibility is doing evything via the NAS IP address. You have then to define the MQTT Container 1883 port to the "outside" world. WIthin Grott you still have to specify the grott ip address but that would be the NAS ip address.

**Reply by Winfryd** [reply]
Hello Johan, I have some succes now. I tried installing via your faq of docker via scheduler instead of Container Manager (docker in synology). The gmqttip= did the trick. I see now in the enviromentals of Container Manager this setting. Still have to figure out to use the grott.ini .
![image](https://github.com/user-attachments/assets/080ff1a5-9d32-457c-beaa-6c10548432bc)

It was that easy.

Now the hard part getting all values I like to see into domoticz. 
I read that you use/used nodered to get it into domoticz, that was a few years back. Do you know if Domoticz can directly use the data now?

Thanks!

**Comment by johanmeijer** [comment]
@Winfryd yes I use nodered for it. I am sure if it is possible to do this directly e.g.  via mqtt. I will have a look into domoticz if there are new integration possibilities.

**Reply by Winfryd** [reply]
Thanks Johan, Just back from a long weekend. Always thankfull for any tips and howto's but will also try to find out things and hope to learn. Would be great if there are new ways of intergration in Domoticz.
